jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartConfig.java

```diff
@@ -51,6 +52,15 @@ public Builder()
         {
         }
 
+        /**
+         * @param location the directory where parts will be saved as files.
+         */
+        public Builder location(String location)
+        {
+            location(ResourceFactory.root().newResource(location).getPath());
```

Comment:
Using the root resource factory to create a resource only to get its path is quite the resource anti-pattern:
you end up doing `Paths.get(location)` with lots of overhead and exposing the risk of leaking a mount if the given location is a `jar:file:` URI.

Reply:
There is a `Paths.get(URI)` too.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Paths.html#get(java.net.URI)

```
jshell> var p1 = Paths.get(URI.create("file:///tmp/foo.txt"))
p1 ==> /tmp/foo.txt

jshell> Files.exists(p1)
$9 ==> false

jshell> Files.writeString(p1, "Hello")
$10 ==> /tmp/foo.txt

jshell> Files.exists(p1)
$11 ==> true
```

Reply:
There's also a `Path.of(URI)` (note the lack of "s")

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/Path.html#of(java.net.URI)

```
jshell> var p2 = Path.of(URI.create("file:///tmp/bar.txt"))
p2 ==> /tmp/bar.txt

jshell> Files.exists(p2)
$13 ==> false

jshell> Files.writeString(p2, "Hello Bar")
$14 ==> /tmp/bar.txt

jshell> Files.exists(p2)
$15 ==> true
```

Reply:
> The problem with Paths.get(String) is that it only handles file paths and not URIs. Resource resolves that and can handle portable configuration

Even if we stick with this awkward code, it shouldn't be using `ResourceFactory.root()`.
Can we just register a properly managed `ResourceFactory` against the `LifeCycle` of something else from here?

Reply:
> so this will throw IAE if there is some strange jar:file URI passed in now.

Seems like a sensible change to avoid mount tracking issues.

> MultiPartConfig should have just taken a resource in the first place!

Totally agree.